src/pubtools/_impl/tracing.py

```diff
@@ -155,7 +193,26 @@ def force_flush(self):
             self._processor.force_flush()
         log.info("Flush trace data into OTEL collectors")
 
-    @property
-    def provider(self):
-        """Trace provider"""
-        return self._provider
+
+class CLITracingWrapper(TracingWrapper):
+    # Clients aren't expected to have open-telemetry. This version provides
+    # pass through functions to make open-telemetry a soft dependency.
+
+    # This class doesn't override reset, as it's intended for testing only.
+
+    def instrument_func(self, span_name=None, carrier=None, args_to_attr=False):
+        def _instrument_func(func):
+            @functools.wraps(func)
+            def wrap(_):
+                pass
```

Comment:
This seems to be broken in two ways:

- `func` is never called, so this decorator replaces every decorated function with a replacement which does nothing.
- `wrap` accepts only a single positional argument.  This means `@instrument_func` will break any function with a signature other than "single positional argument".  The general recipe for a generic decorator is to accept any number of positional and named arguments.

These are severe bugs to the point of not working at all, so if there aren't any tests which caught this, there should be some.

That being said, if you follow my suggestion in the other thread of not introducing a separate class for this, we'd be reusing the existing instrument_func implementation which is already tested.